codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsProtocolUtils.java

```diff
@@ -162,4 +172,88 @@ public static void handleDecodeError(GoWriter writer, String returnExtras) {
     public static void handleDecodeError(GoWriter writer) {
         handleDecodeError(writer, "");
     }
+
+    /**
+     * generateXMLStartElement generates the XML start element for a shape. It is used to generate smithy xml's startElement.
+     *
+     * @param context  is the generation context.
+     * @param shape    is the Shape for which xml start element is to be generated.
+     * @param dst      is the operand name which holds the generated start element.
+     * @param inputSrc is the input variable for the shape with values to be serialized.
+     */
+    public static void generateXMLStartElement(GenerationContext context, Shape shape, String dst, String inputSrc) {
```

Comment:
I think that pattern of spliting utils is good to be followed with go files. Not sure if we want/need to do this with Java classes.

Reply:
Mainly so that they are easier to find, Having a XMLProtocolUtils.java class would make it easier to find XML relevant utilities, instead of seraching a generic AwsProtocolsUtils.  The benefit of this is that these XMlProtocollUtils could be moved to the smithy-go repo as well since they are not AWS specific.

Reply:
I wonder how much we want to expose our XML stuff. The only XML protocols we have are considered deprecated. In the industry as a whole XML hasn't been popular for quite some time. We necessarily have to expose our XML go module for it to work, but I wonder how useful external customers would find it.

All in all I'm +0 on breaking out XML utilities and -0 on having them in smithy-go.